Common.xml: AIRSPEED message

hamishwillee opened the pull request:
Proposed message for AIRSPEED_SENSOR instances, as discussed in #1584. Assumption is sending rate would be low - 1 or 2 Hz, and probably something streamed on request using MESSAGE_INTERVAL rather than "by default".

Has 
- instance: id
- airspeed (IAS)
- temperature
- differential pressure
- static pressure. 

Points for discussion
- Are there any other fields needed? Sensor type?
- Are these the right units? (I copied similar fields in other messages)

FYI @dagar @sfuhrer @RomanBapst @julianoes @auturgy @LorenzMeier 

PS Under current rules I assume we don't merge until there is an implementation, but we can at least approve the design

sfuhrer commented:
I would consider also having fields for calibrated (CAS) and true (TAS) airspeed, as they are what's usually used for control and navigation. In the [VFR_HUD](https://mavlink.io/en/messages/common.html#VFR_HUD) we're leaving it open if the "airspeed" field is corresponding to IAS or CAS, but for a new message I would rather have it clearly determined.

hamishwillee commented:
@sfuhrer The argument for IAS was that a sensor normally puts out IAS, with calculation of CAS or TAS done in the autopilot - so not "AIRSPEED_RAW". I have no opinion personally, other than to say that if we can provide both then that would be fine by me. 

@auturgy thoughts?

sfuhrer commented:
> @sfuhrer The argument for IAS was that a sensor normally puts out IAS, with calculation of CAS or TAS done in the autopilot - so not "AIRSPEED_RAW". I have no opinion personally, other than to say that if we can provide both then that would be fine by me.

Well I think there are also CAN airspeed sensors out there that directly output IAS and TAS. But I guess it would also be fairly easy to add fields for that later once we have a concrete implementation out.

auturgy commented:
A separate message, as this message is intended to reflect sensor readings not estimator output.

Regards,

James 

> On 10 Mar 2021, at 8:06 am, JaeyoungLim <notifications@github.com> wrote:
> 
> ﻿
> One of the missing information that would be nice to have is angle of attack / sideslip angles.
> 
> Would this better fit in a new mavlink message? or can we also include this in Airspeed_raw? @hamishwillee Would you have some insights on this?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

Jaeyoung-Lim commented:
@auturgy I was talking about sensor readings. The ASLUAV dialect has a message that outputs the vane values in mavlink.

hamishwillee commented:
@Jaeyoung-Lim I'd say that if these are actually output *by the sensor* then they are possible candidates for this message. Are they (It isn't clear from [SENSOR_AIRFLOW_ANGLES](https://mavlink.io/en/messages/ASLUAV.html#SENSOR_AIRFLOW_ANGLES) if they are calculated from other data in the autopilot)?

Do we also need sideslip ? What are these values used for?

Jaeyoung-Lim commented:
@hamishwillee I thought the `SENSOR_` prefix means that the information was coming out of a sensor :smile: Yes, they are sensor outputs and these are important information for FW/VTOLs that are flying out of the cruising flight envelop and probably optionally used for system identification.

hamishwillee commented:
@Jaeyoung-Lim Ha ha. The `_RAW` tells you the data is unscaled/not via the autopilot. 

So would the addition be.
```
<field type="float" name="angle_of_attack" units="rad">Angle of attack. NaN if not supported.</field>
<field type="float" name="sidelslip" units="rad">Sideslip angle of attack. NaN if not supported.</field>
```

@auturgy Further comments?

hamishwillee commented:
Discussed in the devcall. In summary, angle_of_attack/sideslip will be synthesized values from an airspeed computer of some kind - so whether or not they are "sensor" they are not "raw". So if these are part of the message there is an objection to the name.

@auturgy Has agreed to propose a message for air computer data (either a replacement for this, or something that co-exists to send the additional information). This will be discussed in the next dev call and some decision made on how to proceed.

hamishwillee commented:
@LorenzMeier @auturgy @julianoes FYI, as discussed in dev call this updates `AIRSPEED` with fields for error and sensor type (enum). The airspeed is documented as CAS if available, otherwise IAS (it is no longer raw sensor data, so that is OK).

I added message/enum to `development.xml` (rather than common.xml) as per new process, and added a comment in common.xml just to note the id reservation.

Any objections to merging?

hamishwillee commented:
OK, as there are no objections I have merged as WIP into development.xml.
